tests/deprecated_methods/tests/deprecated_methods.rs

```diff
@@ -0,0 +1,8 @@
+use std::{fs, path::PathBuf};
+
+#[test]
+fn test() {
+    let path = PathBuf::from(std::env::var("OUT_DIR").unwrap()).join("test.rs");
+    let s = fs::read_to_string(path).unwrap();
+    assert_eq!(s.match_indices("#[deprecated]").count(), 1);
+}
```

Comment:
I would think that just detecting the `#[deprecated]` string seems not enough to test this functionality. Is it possible to verify that this part is generated in the correct place, or to check that the entire generated code matches what is expected?

Reply:
I've oriented myself to the [disable comments test](https://github.com/hyperium/tonic/blob/4c9356c10eb7838e97bbf303f9868aee2364b1af/tests/disable_comments/tests/disable_comments.rs). Testing if `#[deprecated]` is generated above the correct method should be doable by checking `#[deprecated] pub fn somefunction`.

Reply:
```suggestion
#[test]
fn test() {
    let path = PathBuf::from(std::env::var("OUT_DIR").unwrap()).join("test.rs");
    let s = fs::read_to_string(path)
        .unwrap()
        .split_whitespace()
        .collect::<Vec<_>>()
        .join(" ");
    assert!(s.contains("#[deprecated] pub async fn deprecated("));
    assert!(!s.contains("#[deprecated] pub async fn not_deprecated("));
}
```
This is only alternative I can think of atm

Reply:
I am not sure if the way is appropriate for testing code generating, but I feel it can test the attributes.

@djc How do you think?

Reply:
@tottoto not entirely sure what you mean. IMO the [proposed test](https://github.com/hyperium/tonic/pull/1879/files#r1729465879) is good enough for now.

Reply:
> IMO the [proposed test](https://github.com/hyperium/tonic/pull/1879/files#r1729465879) is good enough for now.

Thanks. I was not confident so I wanted other's opinions.